Lib9c/TypedAddress/PledgeAddress.cs

```diff
@@ -0,0 +1,59 @@
+using Bencodex;
+using Bencodex.Types;
+using Libplanet.Crypto;
+
+namespace Nekoyume.TypedAddress
+{
+    public readonly struct PledgeAddress : IBencodable
```

Comment:
If there is a class or struct that is repeated or similar to `AgentAddress`, it would be nice to use the C# Generic feature.

Reply:
What is the example to use C# Generic feature in this case? 🤔 

Maybe like...:

```csharp
public readonly struct TypedAddress<T> : IBencodable;
public readonly AgentAddress : TypedAddress<SomeType>;
```

But C# doesn't allow to inherit struct type. 🤔

Reply:
Aha. My mistake, 😀